.github/workflows/ci.yml

```diff
@@ -66,6 +66,31 @@ jobs:
           -DS3_PRESIGNED_PUT_URL="put-url"
       - name: Build System Tests
         run: make -C build/ help | grep system | tr -d '. ' | xargs make -C build/
+  build-check-install:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone This Repo
+        uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Configure CMake build
+        run: |
+          mkdir installed-headers-dir installed-libs-dir
+          curl https://cmake.org/files/v3.2/cmake-3.2.0-Linux-x86_64.tar.gz -o cmake.tar.gz
+          tar -xf cmake.tar.gz
+          mkdir build && cd build
+          ../cmake-3.2.0-Linux-x86_64/bin/cmake .. \
+          -DBUILD_DEMOS=0 -DBUILD_TESTS=0 \
+          -DCSDK_HEADER_INSTALL_PATH="../installed-headers-dir" -DCSDK_LIB_INSTALL_PATH="../installed-headers-dir"
+      - name: Install libraries and verify install location
+        run: |
+          sudo make -C build/ install
+          while IFS="" read -r p || [ -n "$p" ]
+          do
+            if [[ $p != "$PWD/installed-headers-dir"* ]] && [[ $p != "$PWD/installed-libs-dir"* ]]; then
+              exit 1
+            fi
+          done <build/install_manifest.txt
```

Comment:
It would be good to test the `INSTALL_LIBS` variable as well to check that it only installs the specified libraries

Reply:
Ideally but that is a trickier task... It leads to an SSoT problem where we need to list each header and source of a library then cross-check with what was outputted by the manifest. Anyway, it is unlikely this functionality will change by code changes that are made later on unless `tools/install.cmake` is edited.

Reply:
I have added a comment to `tools/install.cmake` though:
```
When making changes to this file, please verify that -DINSTALL_LIBS="prefix-name" still works.
```

Reply:
>  It leads to an SSoT problem where we need to list each header and source of a library then cross-check with what was outputted by the manifest.

That's a valid point. Instead of checking for each header and library file installation in the target directories, a simpler check of just the presence of the library (passed through the `INSTALL_LIBS` flag) in the file names can be added, along with just a sanity check that no file carries the name of a library that is not expected to be installed.

Reply:
Sure, I've added this check for MQTT;DEFENDER